roles/rclone_serve_restic/tasks/main.yml

```diff
@@ -0,0 +1,40 @@
+---
+
+- name: Ensure user is present
+  ansible.builtin.user:
+    name: "{{ rclone_serve_restic_user }}"
+    state: present
+    system: true
+  register: rclone_serve_restic_user_res
+
+- name: Ensure config directory is present
+  ansible.builtin.file:
+    path: "{{ rclone_serve_restic_config_path }}"
+    state: directory
+    mode: '0640'
+    owner: "{{ rclone_serve_restic_user_res.uid }}"
+    group: "{{ rclone_serve_restic_user_res.group }}"
+
+- name: Ensure config is present
+  ansible.builtin.template:
+    src: to-ini.j2
+    dest: "{{ rclone_serve_restic_config_file }}"
+    mode: '0600'
+    owner: "{{ rclone_serve_restic_user_res.uid }}"
+    group: "{{ rclone_serve_restic_user_res.group }}"
+  vars:
+    ini: "{{ rclone_serve_restic_backend_config_merged }}"
+
+- name: Ensure container is started
+  community.docker.docker_container:
+    image: "{{ rclone_serve_restic_container_image }}"
+    name: "{{ rclone_serve_restic_container_name }}"
+    state: started
+    pull: true
```

Comment:
pull in a seperate task with the `community.docker.docker_image` module - this allows tasks which rely on networking to be tagged and/or retried (`while`/`until`...)